src/Coordination/KeeperServer.cpp

```diff
@@ -367,6 +368,12 @@ void KeeperServer::shutdownRaftServer()
     }
 
     raft_instance->shutdown();
+
+    keeper_context->server_state = KeeperContext::Phase::SHUTDOWN;
+
+    if (create_snapshot_on_exit)
+        raft_instance->create_snapshot();
```

Comment:
This call is synchronous? It's strange that I cannot find it in `nuraft::raft_server`.

Reply:
did you update submodule? it's in `handle_commit.cxx`

it is synchronous.

Reply:
Really!